python/lsst/ts/FiberSpectrograph/dataManager.py

```diff
@@ -0,0 +1,130 @@
+# This file is part of ts_FiberSpectrograph.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["SpectrographData", "DataManager"]
+
+import dataclasses
+import os.path
+
+import astropy.time
+import astropy.io.fits
+import numpy as np
+
+
+@dataclasses.dataclass
+class SpectrographData:
+    """Class to hold data and metadata from a fiber spectrograph.
```

Comment:
Does this include all the things that you set which controls data acquisition?

Reply:
I'm not sure what you mean. The only things that matter are the number of pixels (fixed for a given spectrograph) and the duration of the exposure. I've not implemented sub-frame readout, and I don't think it would be necessary or desirable (readout time is a handful of milliseconds, so subframes would just make downstream processing harder).

There was talk of doing an "exposure loop" (take 10 exposures of the same duration), but whether that is implemented at the CSC level or in the controller object hasn't been determined, so we don't know what metadata we'd need to write out for it.